contracts/CourtPresaleActivate.sol

```diff
@@ -49,10 +59,93 @@ contract CourtPresaleActivate is IsContract, ApproveAndCallFallBack {
         require(_token == address(presale.contributionToken()), ERROR_WRONG_TOKEN);
 
         // move tokens to this contract
-        require(ERC20(_token).safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
+        ERC20 token = ERC20(_token);
+        require(token.safeTransferFrom(_from, address(this), _amount), ERROR_TOKEN_TRANSFER_FAILED);
 
+        _buyAndActivate(_from, _amount, _token);
+
+        // refund leftovers if any
+        _refund(token);
+    }
+
+    /**
+    * @dev This function needs a previous approval on the external token used for the contributed amount.
+    *      It will pull the approved tokens, convert them in Uniswap to the presale contribution token,
+    *      convert the obtained tokens in the presale instance, and activate the converted tokens in the
+    *      jurors registry instance of the Aragon Court.
+    * @param _token Address of the external contribution token used
+    * @param _amount Amount of contribution tokens to be converted and activated
+    * @param _minTokens Minimum amount of presale contribution tokens obtained in Uniswap
+    * @param _minEth Minimum amount of ETH obtained in Uniswap (Uniswap internally converts first to ETH and then to target token)
+    * @param _deadline Transaction deadline for Uniswap
+    */
+    function contributeExternalToken(
+        address _token,
+        uint256 _amount,
+        uint256 _minTokens,
+        uint256 _minEth,
```

Comment:
Not particularly opposed to exposing `minTokens` and `minEth` like this, but I'm wondering if we should.

`minEth`, if I understand it correctly, is to control slippage across exchanges (but I don't understand why it matters, since you're controlling the end token amount anyway. Maybe for liquidity reasons?). I don't think there's much point in exposing this parameter and we could likely just use `1` where it's needed.

`minTokens` is confusing, because we have three tokens involved: external, contribution, and bonded. As a user, I'd likely be most interested in the final bonded token amount I'm getting, so it may be more useful (and less confusing) to provide that as a parameter instead.

Reply:
This is just forwarding the Uniswap interface:
https://docs.uniswap.io/smart-contract-api/exchange#tokentotokenswapinput
I think it makes sense to leave it as is, because in the end to understand you have to go to Uniswap docs/code, and adding our own intermediate translation would only create more confusion in my opinion.
About `minEth`, we can always make frontend to fill in a `1` in it, so I don't see any problem in having this param.